api/app.rb

```diff
@@ -232,15 +255,35 @@ class RenderApp < Sinatra::Base
 
   # TODO: The :id should be parsed against the same regex as above
   post '/:id' do
-    answers = params.to_json
-    cmd = FlightJobScriptAPI::SystemCommand.flight_create_script(params[:id], user: @current_user, stdin: answers)
+    hash = params.to_h.dup
+    # Ensure name and notes are stringss and treat nil/empty-string as the same input
+    # NOTE: This prevents a question from using the 'name'/'notes' keys
+    #       Consider refactoring to be a standard JSON:API route
+    name = hash.delete('name').to_s
+    name = nil if name.empty?
+    notes = hash.delete('notes').to_s
+    notes = nil if notes.empty?
+    answers = hash.to_json
```

Comment:
I have deviated slightly from the requested implementation because the purposes for splitting the `payload` syntax:
* Allow the question ids `name` and `node`, and
* Prevent the proliferation of "reserved keys" as features are added.

The new syntax handles this well and has been implemented as requested as the "Structured" format.

The old syntax however still treats `name` and `node` as reserved keys which can not be used. This implies all future reserved keys should also be black-listed, which reduces backwards compatibility.

Instead the old "Unstructured" syntax treats all provided keys as `answers`. This means it can handle question ids `node` and `name`. It also means the addition of future reserved keys will not affect the "Unstructured" request syntax.

This means the old "Unstructured" requests will not support the additional features. All new features will have to be done by the "Structured" request syntax. See the [docs](docs/routes.md) for the full specification.

NOTE: This does mean that `answers` is a reserved key which can not be used with the "Unstructured" syntax as it will toggle the format to "Structured". However this limitation still prevents reserved key proliferation. An alternative would be to use a custom `Content-Type` as the toggle OR bump the API version and break backwards compatibility.

Reply:
The implementation you've gone with is fine.  We'll likely drop support for the old "unstructured" format in the next release.  Doing things this was does make developing the client against a moving api easier.